bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/config/CredentialFactory.java

```diff
@@ -91,16 +92,9 @@ private static synchronized HttpTransport getHttpTransport()
    * GCE VM. See: <a href="https://developers.google.com/compute/docs/authentication"
    * >Authenticating from Google Compute Engine</a>.
    */
-  public static Credentials getCredentialFromMetadataServiceAccount()
+  public static OAuth2Credentials getCredentialFromMetadataServiceAccount()
       throws IOException, GeneralSecurityException {
-    Credentials cred = new ComputeEngineCredentials(getHttpTransport());
-    try {
-      cred.refresh();
-    } catch (IOException e) {
-      throw new IOException("Error getting access token from metadata server at: " +
-          ComputeCredential.TOKEN_SERVER_ENCODED_URL, e);
-    }
-    return cred;
+    return new ComputeEngineCredentials(getHttpTransport());
```

Comment:
NOTE to the reviewer: The refresh now happens asynchronously in the OAuth2CredentialsInterceptor